operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -51,7 +51,7 @@ public void start() {
       }
     } catch (Exception e) {
       log.error("Error retrieving the server version. Exiting!", e);
-      System.exit(1);
+      throw new OperatorException("Error retrieving the server version");
```

Comment:
We should add the root cause exception, are we sure that this is always version?

Reply:
It most often is, but, strictly speaking, there might be other causes.

Reply:
chained the thrown exception to the new one, would this work the way you expect?